### ISSUE TITLE
Improve how we detect that a post has been edited

### ISSUE THREAD

**Issue by Tug** [wordpress-mobile/gutenberg-mobile#266]
Currently we detect if a post is modified by checking if the resulting html changed:

See:
https://github.com/wordpress-mobile/gutenberg-mobile/pull/225
https://github.com/wordpress-mobile/gutenberg-mobile/pull/242
https://github.com/wordpress-mobile/gutenberg-mobile/pull/264

We'd like to improve on that logic and reuse the `isEditedPostDirty` selector from the gutenberg store.

However there are a few road blocks along the way:

- `parse` and `serialize` of blocks isn't always symmetric. For instance:
```js
> serialize( parse( `<!-- wp:heading {"align":"center"} -->
<h2 style="text-align:center;">Welcome to Gutenberg!</h2>
<!-- /wp:heading -->` ) )
`<!-- wp:heading {"align":"center"} -->
<h2 style="text-align:center">Welcome to Gutenberg!</h2>
<!-- /wp:heading -->`
```
(notice the `;` at the end of `text-align:center`)

- Some blocks are updated when the post loads. For instance `paragraph` and `heading` both use the `RichText` component and modify the attributes in `onContentSizeChange`. But `onContentSizeChange` is called on load by `RCTAztecView` (RichText underline component).

**Comment by Tug** [comment]
Yes, it's very close. I suggest we close #227 though as we don't use a hash anymore to check that something has changed

**Comment by SergioEstevao** [comment]
@Tug  is this something that we still want to pursue?

**Comment by hypest** [comment]
We'd still want to have a performant way of checking for content dirtiness so, marking this as an enhancement and adding it to the Backlog.
